docs/solana/migration/exchange.mdx

```diff
@@ -0,0 +1,78 @@
+
+:::
+
+Helium will be transitioning from its own L1 to Solana. The proceedure for Helium should be the same
+as the proceedure for all other Solana SPL Tokens.
+
+All the tokens are on the `TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA` program.
+
+The mint IDs are as follows:
+
+| Name   | ID                                          |
+| ------ | ------------------------------------------- |
+| HNT    | hntyVP6YFm1Hg25TN9WGLqM12b8TQmcknKrdu1oxWux |
+| MOBILE | mb1eu7TzEc71KxDpsmsKoucSSuuoGLv1drys1oP2jh6 |
+| IOT    | iotEVVZLEywoTn1QdwNPddxPWszn3zFhEot3MfL9fns |
+| DC     | dcuc8Amr83Wz27ZkQ2K9NS6r8zRpf1J6cvArEBDZDmm |
+
+These mint IDs exist both on `mainnet-beta` and `devnet`, so you can test.
+
+## Wallet Migration
+
+To migrate wallet(s) to Solana, you can use the migration service. Instead of sending all
+transactions to inflate helium state on Solana during the transition period, transactions are
+instead processed as needed. When a user opens their wallet in the helium wallet app, it requests
+our migration service to get the needed transactions, then submits them to Solana.
+
+Since exchange wallets may not use the wallet app, you can use the service directly:
+
+For devnet:
+
+`https://migration.web.test-helium.com/migrate/*solana-wallet-address*?limit=1000&offset=0`
+
+This will get the list of serialized transactions, which should be sent to Solana.
+
+To get the Solana wallet address from a helium wallet address, call:
+
+`https://migration.web.test-helium.com/helium/*helium-wallet-address*`
+
+You can also use our command line utility from
+[here](https://github.com/helium/helium-program-library/releases/tag/v0.0.27-migration-cli)
+
+```
+$ env MIGRATION_SERVICE_URL=https://migration.web.test-helium.com env SOLANA_URL=https://api.devnet.solana.com env SOLANA_WSS_URL=wss://api.devnet.solana.com ./migration-tx-executor --wallet *your-solana-wallet*
+```
+
+You can see typescript code demonstrating how to use this
+[here](https://github.com/helium/helium-program-library/blob/master/packages/migration-service/src/test-submit.ts#L37)
+
+## Trustless Wallet Migration
+
+While the above tools are convenient, they rely on our migration service. Luckily, the on-chain
+implementation of the migration is trustless. The full state of the helium chain was converted to a
+list of Solana transactions and then compressed into a merkle root on Solana. This means that anyone
+can execute the transactions to inflate a wallet so long as they provide a proper proof that the
+provided transactions are valid. Each transaction is only allowed to be executed once.
+
+We can see one such transaction here:
+
+https://explorer.solana.com/tx/5Y8vfMAfhyqgXsp1YnvPVHMdK4PUryLqMCgGynQXg9tPkUVehK438XviXz3ciSkg4PhqMZ8UyrSSVj7pgY8uq1xr?cluster=devnet
+
+If you would like a copy of the database serving the proofs for the migration service, please
+contact the helium foundation. You can then run your own instance of the
```

Comment:
```suggestion
[contact the Helium Foundation](hello@helium.foundation). You can then run your own instance of the
```

Reply:
We should probably also just publish the proofs in shadow or arweave and link here.